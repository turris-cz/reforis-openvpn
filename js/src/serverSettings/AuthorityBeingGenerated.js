/*
 * Copyright (C) 2019 CZ.NIC z.s.p.o. (http://www.nic.cz/)
 *
 * This is free software, licensed under the GNU General Public License v3.
 * See /LICENSE for more information.
 */

import React, { useEffect } from "react";
import PropTypes from "prop-types";

import { useWSForisModule, Spinner } from "foris";

AuthorityBeingGenerated.propTypes = {
    ws: PropTypes.object.isRequired,
    onReload: PropTypes.func.isRequired,
};

export default function AuthorityBeingGenerated({ ws, onReload }) {
    const [generateCA] = useWSForisModule(ws, "openvpn", "generate_ca");
    useEffect(() => {
        if (generateCA && generateCA.status === "succeeded") {
            onReload();
        }
    }, [generateCA, onReload]);

    return (
        <>
            <h3>{_("Generating certification authority")}</h3>
            <Spinner className="my-3 text-center" />
            <p>{_("Your certification authority is now being generated. It usually takes a few minutes. Settings will appear here automatically once the authority is ready.")}</p>
        </>
    );
}
