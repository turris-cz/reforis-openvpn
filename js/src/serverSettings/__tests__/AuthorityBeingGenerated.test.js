/*
 * Copyright (C) 2019 CZ.NIC z.s.p.o. (http://www.nic.cz/)
 *
 * This is free software, licensed under the GNU General Public License v3.
 * See /LICENSE for more information.
 */

import React from "react";
import { render, act } from "customTestRender";

import { WebSockets } from "foris";

import AuthorityBeingGenerated from "../AuthorityBeingGenerated";

describe("<AuthorityBeingGenerated />", () => {
    const handleReload = jest.fn();
    let webSockets,
        componentContainer;

    beforeEach(() => {
        webSockets = new WebSockets();
        const { container } = render(<AuthorityBeingGenerated ws={webSockets} onReload={handleReload} />);
        componentContainer = container;
    });

    it("should match snapshot", () => {
        expect(componentContainer).toMatchSnapshot();
    });

    it("should call onReload when CA is ready", () => {
        expect(handleReload).not.toBeCalled();
        const wsMessage = { module: "openvpn", action: "generate_ca", data: {status: "succeeded"} };
        act(() => webSockets.dispatch(wsMessage));
        expect(handleReload).toHaveBeenCalledTimes(1);
    });
});
