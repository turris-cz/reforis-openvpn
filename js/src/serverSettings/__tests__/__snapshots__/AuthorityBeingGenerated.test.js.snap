// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AuthorityBeingGenerated /> should match snapshot 1`] = `
<div>
  <h3>
    Generating certification authority
  </h3>
  <div
    class="spinner-wrapper my-3 text-center"
  >
    <div
      class="spinner-border "
      role="status"
    >
      <span
        class="sr-only"
      />
    </div>
    <div
      class="spinner-text"
    />
  </div>
  <p>
    Your certification authority is now being generated. It usually takes a few minutes. Settings will appear here automatically once the authority is ready.
  </p>
</div>
`;
